pkg/apimachinery/validation/validation_test.go

```diff
@@ -0,0 +1,226 @@
+package validation_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	k8sValidation "k8s.io/apimachinery/pkg/util/validation"
+
+	"github.com/grafana/grafana/pkg/apimachinery/validation"
+)
+
+func TestValidation(t *testing.T) {
+	// We are not using the out-of-the-box "isQualifiedName" because it allows slashes
+	rsp := k8sValidation.IsQualifiedName("hello/world")
```

Comment:
@filewalkwithme -- the `IsQualifiedName` check for URL parts, so we can not use it directly 😢 

the other options are close but not exactly what we need, so i just copied the regex